api/v1beta1/csiaddonsnode_types.go

```diff
@@ -0,0 +1,104 @@
+/*
+Copyright 2021 The Kubernetes-CSI-Addons Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// CSIAddonsNodeState defines the state of the operation.
+type CSIAddonsNodeState string
+
+const (
+	// Connected represents the Connected state.
+	CSIAddonsNodeStateConnected CSIAddonsNodeState = "Connected"
+
+	// Failed represents the Connection Failed state.
+	CSIAddonsNodeStateFailed CSIAddonsNodeState = "Failed"
+)
+
+type CSIAddonsNodeDriver struct {
+	// Name is the name of the CSI driver that this object refers to.
+	// This must be the same name returned by the CSI-Addons GetIdentity()
+	// call for that driver. The name of the driver is in the format:
+	// `example.csi.ceph.com`
+	Name string `json:"name"`
+
+	// EndPoint is url that contains the ip-address to which the CSI-Addons
+	// side-car listens to.
+	EndPoint string `json:"endpoint"`
+
+	// NodeID is the ID of the node to identify on which node the side-car
+	// is running.
+	// +optional
```

Comment:
@yati1998 @nixpanic if this is optional how the controller can reach the driver on the particular node?